Include all the allowed types for array indices. (#68)
AleoHQ/welcome: documentation/developer/language/04_arrays_and_tuples.md

## Array Indexing

> Access an element of an array starting at index 0. Index must be a `u8`, `u16`, or `u32`.

Reviewer: ```suggestion
Access an element of an array starting at index 0. Indices must be a `u8`, `u16`, or `u32`.
```

## Array Slicing
Access a subset of an array. 

> From index must be a `u8`, `u16`, or `u32`. Omitting from index defaults to index 0.

Reviewer: ```suggestion
The `from` index must be a `u8`, `u16`, or `u32`. Omitting from index defaults to index 0.
```

> The `from` index must be a `u8`, `u16`, or `u32`. Omitting from index defaults to index 0.

Reviewer: ```suggestion
The `from` index must be a `u8`, `u16`, or `u32`. Omitting the `from` index will default to index 0.
```

> To index must be a `u8`, `u16`, or `u32`. Omitting to_index defaults to the length of the array minus one.

Reviewer: ```suggestion
The `to` index must be a `u8`, `u16`, or `u32`. Omitting the `to` index defaults to the length of the array.
```
